### PR TITLE
Exploration of further Biological and Clinical features

### PR DESCRIPTION
**Pull request by mcreixell**
CIMP-2 low, CIMP-1 intermediate, and CIMP+ high. Cluster 22 shows a very slight inverse correlation with amount of methylation, and cluster 14 shows a much higher p-site abundance in CIMP-2 than the other two groups. Looks like similar to staging, there are no very large phosphorylation changes by methylation status, and when there are differences the pattern is not a constant increase/decrease other than with the aforementioned clusters... @aarmey 

meth

### CODE REVIEW

**Comment by mcreixell** [comment]
@aarmey The last two clinical features I'd try are smoker/non-smoker and dominant histological subtype. However, I'm not sure if the latter has enough patients in each category:

[hist_subtype.xlsx](https://github.com/meyer-lab/resistance-MS/files/5668669/hist_subtype.xlsx)

**Comment by mcreixell** [comment]
Alternatively, we could also explore clusters tracking wit hot/cold tumors or enrichment of specific immune cell types:

[ImmuneColdHot_status.xlsx](https://github.com/meyer-lab/resistance-MS/files/5668737/ImmuneColdHot_status.xlsx)

**Comment by aarmey** [comment]
What would these predictions provide us that we don't already have? Unless there is a *very* clear answer to this question, I don't think more is better. I guess immune infiltration would suggest different treatment approaches. Smoking doesn't seem very useful. Histology sounds risky due to the varying numbers.

**Comment by mcreixell** [comment]
I didn't see any difference in smokers versus non-smokers and didn't even try histology subtypes because there were multiple subtypes with only 1-2 patients and the majority were acinar. I do see big differences in the immune infiltration:

Hot vs Cold only:
HotCold_MW


NAT vs Hot vs Cold:
HotCold_Kusk

**Comment by aarmey** [comment]
Is your test here [NAT vs. cold vs. hot] or just [cold vs. hot]? It seems like the former...

**Comment by mcreixell** [comment]
In the first plot the test is [cold vs hot], in the second panel the test is [NAT vs cold vs hot].

**Comment by aarmey** [comment]
But which way is the statistical test done?

**Comment by mcreixell** [comment]
Mann Whitney in the first plot and Kuskal in the rest. I’m not sure if this answers your question?

**Comment by aarmey** [comment]
Am wondering which is used for the p-values—I'm guessing MW? If so, that's right. It's still surprising that every cluster would be significant...

**Comment by mcreixell** [comment]
The way I calculate the p-values of the first boxplot and is using MW:

https://github.com/meyer-lab/resistance-MS/blob/ded7d1b3125d0bcd347817b2c69b59d193591511/msresist/figures/figureM3.py#L96-L106

Whereas the p-values of the panel of boxplots comparing NATs, cold tumors, and hot tumors, is using Kuskal:

https://github.com/meyer-lab/resistance-MS/blob/ded7d1b3125d0bcd347817b2c69b59d193591511/msresist/figures/figureM5.py#L62-L77

**Comment by aarmey** [comment]
Ok. I don't think including the NAT is useful, so I'd remove that.

**Comment by mcreixell** [comment]
Ok. Do you think it would be worth looking into specific immune cell type populations? It’d be cool to make an entire figure on immune features. 

> Alternatively, we could also explore clusters tracking with hot/cold tumors or enrichment of specific immune cell types:
> 
> [ImmuneColdHot_status.xlsx](https://github.com/meyer-lab/resistance-MS/files/5668737/ImmuneColdHot_status.xlsx)

**Comment by mcreixell** [comment]
On the notebook `mlr.ipynb` I started exploring PLSR/Lasso to predict the infiltration scores of the immune populations showed in the PCA below. However, the prediction results are not very encouraging so maybe it's not worth pursuing this regression any further? If so, what I would have in mind for figureM5 is the MW tests between [cold vs. hot], logistic regression ROC plot, relevant cluster motifs, and upstream kinases. @aarmey 

Screen Shot 2020-12-13 at 10 05 44 PM
Screen Shot 2020-12-13 at 10 05 54 PM
Screen Shot 2020-12-13 at 10 06 02 PM

**Comment by mcreixell** [comment]
Never mind... I found out why all all the MW tests were significant... Silly mistake. Turns out I was testing [NAT vs Cold] instead of [Cold vs Hot] (index error) but I was showing the correct p-site abundance data in the box plot (cold vs hot). Just fixed this and despite some differences in p-signal, none of the tests were significant after Bonferroni correction (no asterisks):

Screen Shot 2020-12-13 at 10 45 19 PM

Given this result, I would just focus on refining what we have. I think that FiguresM1, 3&4 are close, and FigureM2 may need a bit more work—I will open an issue to discuss this figure—but the point of this figure is more or less also clear so I can start working on the text as well. I will also be working on doing the same predictions as in figuresM3&4 using unclustered / kmeans / gmm only data. I will also think of ways we can compare our upstream kinase predictions to KSEA/Scansite/KinomeXplorer predictions. @aarmey

**Comment by mcreixell** [comment]
(I've deleted the `mlr.ipynb` and will delete figureM5 if you agree with what I said above. After that, this PR contains edits corresponding to hypothesis testing that I want to add to master so I would still like to merge this branch.)

**Comment by aarmey** [comment]
Agreed that the hot/cold doesn't look promising. But isn't a 0.5 r2 not bad for the infiltration? PLSR is the wrong kind of model, but Lasso is fine.

**Comment by aarmey** [comment]
I agree that we should not be exploring new things to add. *Maybe* include the infiltration, but then we should finalize this with the existing analysis. I'll review the code right now.

File: msresist/data/MS/CPTAC/cold_hot.csv [hunk 1]
```diff
@@ -0,0 +1,212 @@
+﻿Sample ID,Group,
```
Comment (aarmey) [inline msresist/data/MS/CPTAC/cold_hot.csv:1]: Don't think you need this anymore?

**Review by aarmey — APPROVED** [review]
One small suggestion. Otherwise this looks fine to me.

File: msresist/data/MS/CPTAC/cold_hot.csv (hunk 1 above)
Comment (mcreixell) [reply msresist/data/MS/CPTAC/cold_hot.csv]: done.

**Comment by mcreixell** [comment]
This is what figureM5 currently looks like:

image

The tests are failing due to the following error: 

Screen Shot 2020-12-15 at 12 14 52 AM

I checked and this error doesn't occur on Aretha. However, there is a warning about the argument it's complaining about `check_pickle`:
Screen Shot 2020-12-15 at 12 16 05 AM

Do you know what's going on? @aarmey

**Comment by aarmey** [comment]
I think you can fix this by pinning the joblib version in `requirements.txt`. Add the line `joblib==0.17.0`.

**Comment by mcreixell** [comment]
Ok this worked, thanks. I'm going to merge this.

**Pull request merged**